### PR TITLE
adds abstraction for connection

### PR DESCRIPTION
**Pull request by tomaszduda23**
After that change any other protocol can be used for transport easily.

### CODE REVIEW

**Comment by mdevaev** [comment]
Did you test it with avr? I wouldn't want to accept this without any testing.

**Comment by tomaszduda23** [comment]
I tested on stm32 only. I have no access to avr right now.

**Comment by mdevaev** [comment]
Okay, I'll test it myself but not right now.

**Comment by tomaszduda23** [comment]
I will take a look soon. Maybe you would be interested https://github.com/esphome/feature-requests/issues/1918

**Comment by tomaszduda23** [comment]
It works over restapi and websocket so it would be possible to integrate. If it gets to the main.

File: hid/platformio-avr.ini [hunk 1]
```diff
@@ -11,6 +11,7 @@ lib_deps =
 	git+https://github.com/Harvie/ps2dev#v0.0.3
 	digitalWriteFast@1.0.0
 	HID@1.0
+	SPI
```
[the remark this answers is not in the document]
Comment (tomaszduda23) [reply hid/platformio-avr.ini:14]: It was added automatically for `make spi`. You need this or lib_ldf_mode = deep

Dependency Graph
|-- HID-Project @ 2.8.2+sha.afae388
|   |-- HID @ 1.0
|-- ps2dev @ 0.0.3+sha.a043002
|-- digitalWriteFast @ 1.0.0
|-- HID @ 1.0
|-- drivers-avr
|   |-- drivers
|   |   |-- digitalWriteFast @ 1.0.0
|   |   |-- SPI @ 1.0
|   |-- ps2dev @ 0.0.3+sha.a043002
|   |-- HID-Project @ 2.8.2+sha.afae388
|   |   |-- HID @ 1.0
|-- drivers
|   |-- digitalWriteFast @ 1.0.0
|   |-- SPI @ 1.0
Building in release mode

**Comment by mdevaev** [comment]
Sorry, it was very long.

**Pull request merged**